presentation-compiler/src/test/scala/org/alephium/ralph/lsp/pc/search/TestCodeProvider.scala

```diff
@@ -81,6 +82,18 @@ object TestCodeProvider {
       searchSettings = settings
     )
 
+  def goToDefinitionSoft(settings: GoToDefSetting = testGoToDefSetting)(code: String): List[(URI, LineRange)] =
+    goTo[SourceCodeState.IsParsed, (SoftAST.type, GoToDefSetting), SourceLocation.GoToDefSoft](
+      code = code,
+      searchSettings = (SoftAST, settings)
+    )
+
+  /** Executes go-to-definition providers for both StrictAST and [[SoftAST]] */
+  def goToDefinition(settings: GoToDefSetting = testGoToDefSetting)(code: String): List[(URI, LineRange)] = {
+    goToDefinitionStrict(settings)(code)
+    goToDefinitionSoft(settings)(code)
```

Comment:
both are executed, but only the results of the soft one are returned, is it expected or you miss a `++ `?

Reply:
`++` is not required. When a go-to-def test is executed on both `Strict` and `Soft` ASTs, they both should return the same result. Running the test already asserts this, but I've made this explicit by checking it in this function.